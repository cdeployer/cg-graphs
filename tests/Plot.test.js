import PlotlyGraph from '../src/components/Plot';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('PlotlyGraph component', () => {
  it('renders a spinner when data is not yet fetched', () => {
    const wrapper = shallow(<PlotlyGraph graphID={'gb/1'} data={[]} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it('does not render a spinner when data is fetched', () => {
    const wrapper = shallow(<PlotlyGraph graphID={'gb/1'} data={[{x:1,y:1}]} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
