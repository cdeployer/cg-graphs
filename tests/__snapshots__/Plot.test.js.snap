exports[`PlotlyGraph component does not render a spinner when data is fetched 1`] = `
<div
  id="gb/1" />
`;

exports[`PlotlyGraph component renders a spinner when data is not yet fetched 1`] = `
<div
  className="plot-div"
  id="gb/1">
  <div
    className="graphSpinner">
    <Loader
      color="#00D49A"
      loading={true}
      size="40px" />
  </div>
</div>
`;
